File: ruby/src/Structural/Bridge/SodaImpSingleton.rb

#//SodaImpSingleton - a Singleton to hold the current SodaImp

class SodaImpSingleton

	@@instance = SodaImpSingleton.new


	def initialize(sodaimp)
		@sodaimp = sodaimp
	end

	def self.instance
		return @@instance
	end

	def getTheSodaImp()
 		return @sodaimp
	end

	def setTheSodaImp(si)
 		@sodaimp = si
	end

	private_class_method :new
end

File: csharp/src/Structural/Bridge/SodaImpSingleton.cs
namespace Structural.Bridge {


//SodaImpSingleton.java - a Singleton to hold the current SodaImp

public class SodaImpSingleton {  
	
   private SodaImp sodaImp;
	public static SodaImpSingleton instance = new SodaImpSingleton();
   
    private SodaImpSingleton() { }
    
    public static SodaImp getSodaImp() {
        return instance.sodaImp;
    }
    public static void setSodaImp(SodaImp si) {
		instance.sodaImp = si;
	}
}

}
